CMakeLists.txt

```diff
@@ -92,15 +92,22 @@ message("-- CMAKE_SYSTEM:           ${CMAKE_SYSTEM}")
 message("-- CMAKE_SYSTEM_VERSION:   ${CMAKE_SYSTEM_VERSION}")
 message("-- CMAKE_BUILD_TYPE:       ${CMAKE_BUILD_TYPE}")
 message("-- TARGET_ARCH:            ${TARGET_ARCH}")
+message("-- CMAKE_CXX_COMPILER_ID:  ${CMAKE_CXX_COMPILER_ID}")
 
 include(tools/ParseOsRelease.cmake)
 
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+set(WARN_FLAGS "/W4 /WX")
```

Comment:
One minor piece of feedback here: when you add `/W4`, it may be different than the default flag CMake decided to pass for whatever build configuration. You'd be getting a warning, like `Command line warning D9025: overriding '/W3' with '/W4'` .. https://stackoverflow.com/questions/58708772/cmake-project-in-visual-studio-gives-flag-override-warnings-command-line-warnin ... that's because by default the flag for `/W3` is already there. So the more elegant solution could have been the one mentioned in the StackOverflow answer `string(REGEX REPLACE "/W[3|4]" "/w" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")`